scipy/stats/tests/test_stats.py

```diff
@@ -7917,6 +7917,222 @@ def test_alias(self):
+    confidence = 0.25*np.random.rand(m)+0.75    # confidence in [0.75,1)
+
+    def test_quantile_test_iv(self):
+        
+        x = [1,2,3]
+        p = 0.5
+        q = 0
+        alternative = 'less'
+
+        message = "`x` must be a one-dimensional array of numbers."
+        with pytest.raises(ValueError, match=message):
+            stats.quantile_test([[1,2,3]],q,p,alternative=alternative)
+        
+        message = "`q` must be a scalar."
+        with pytest.raises(ValueError, match=message):
+            stats.quantile_test(x,[1,2],p,alternative=alternative)
+        
+        message = "`p` must be a float strictly between 0 and 1."
+        with pytest.raises(ValueError, match=message):
+            stats.quantile_test(x,q,[0.5,0.75],alternative=alternative)
+        with pytest.raises(ValueError, match=message):
+            stats.quantile_test(x,q,2,alternative=alternative)
+        with pytest.raises(ValueError, match=message):
+            stats.quantile_test(x,q,-2,alternative=alternative)
+        
+        message = "alternative not recognized; \n"
+        "must be 'two-sided', 'less' or 'greater'"
+        with pytest.raises(ValueError, match=message):
+            stats.quantile_test(x,q,p,alternative='one-sided')
+        
+        message = "`confidence_level` must be a number between 0 and 1."
+        with pytest.raises(ValueError, match=message):
+            stats.quantile_test(x,q,p,alternative=alternative
+                                ).confidence_interval(1)
+
+    def test_confint_naive(self):
+        # Test the confidence intervals against a naive implementation
+        x, m, N, quantile_def, confidence = (self.x, self.m, self.N, 
+                                             self.quantile_def,
+                                             self.confidence)
+
+        def confint_naive(N, quantile, confidence):
+            '''
+            Naive implementation for computing the index of a one-sided CI
+            (lower-bound)
+            '''
+            # compute all probabilities from the binomial distribution for the
+            # quantile of interest
+            bd = stats.binom(N, quantile)
+            ppm = [np.maximum(1-x, 0.0)
+                   for x in np.cumsum([bd.pmf(k) for k in range(N)])]
+
+            # search the index defining a lower-bound for p_work
+            if ppm[0] < confidence:
+                return np.nan
+            else:
+                for k in range(N):
+                    # search for first index reaching below the desired
+                    # confidence
+                    if ppm[k] < confidence:
+                        # lower-bound is the previous index
+                        return k-1
+                # if we exit the loop, then the last index is 
+                # the best possible bound
+                return N-1
+
+        # Test for the one-sided case
+        # .. Derive the lower and upper bounds using naive implem
```

Comment:
Ideally, the naive implementation should ideally be simpler than the original code, but this is much more difficult to follow. 
It sounded like the `binomtest` version I had was correct except for edge cases. Can we consider using that as the naive implementation?

Reply:
I don't have any strong feelings about this. This "naive" implementation was the very first one I had for the computation of the CI. I used it as a test following your suggestion; if you think it's not worth it, we can remove it, all good with me.

Reply:
Yeah, I'll figure something else out.